Synth drivers/eSpeak NG internal: pass in a NULL pointer (path string) when obtaining eSpeak NG version string

josephsl opened the pull request:
Hi,

Needed for Python 3.8, although it is equally applicable to Python 3.7 and makes NVDA code more compliant with eSpeak NG API:

### Link to issue number:
None

### Summary of the issue:
eSpeak NG fails to load on Python 3.8 version of NVDA, traced to espeak.dll::espeak_Info function not receiving path string.

### Description of how this pull request fixes the issue:
Pass in NULL (None) when eSepak NG asks for path string in sepak_Info function because what NVDA is interested in is eSpeak NG version string.

### Testing performed:
Tested under source code and via a private binary copy:

* Success in Python 3.7 version
* Success in Python 3.8 version

### Known issues with pull request:
None

### Change log entry:
None

### Impact:
This is one of the major changes required to move NVDA to Python 3.8, especially given that eSpeak NG will be set as default speech synthesizer on Windows releases other than 10.

Thanks.

josephsl committed:
Synth drivers/eSpeak NG internal: update copyright header

josephsl committed:
Synth drivers/eSpeak NG internal: passin NULL (None) when obtaining version info.

Python 3.8: without passing in a path string, access violation is thrown, which can cause NVDA executable to hang when trying to obtain eSpeak NG version string. Therefore pass in NULL (None) because what NVDA is interested in is synthesizer version.

lukaszgo1 commented on source/synthDrivers/_espeak.py:
```diff
@@ -1,9 +1,9 @@
-#Copyright (C) 2007-2017 NV Access Limited, Peter Vágner
-#This file is covered by the GNU General Public License.
-#See the file COPYING for more details.
+# synthDrivers/_espeak.py
```
Please remove this line to comply with the standard for copyright headers.

lukaszgo1 commented on source/synthDrivers/_espeak.py:
```diff
@@ -369,7 +369,8 @@ def terminate():
 def info():
-	return espeakDLL.espeak_Info()
+	# Python 3.8: a path string must be specified, a NULL is fine when what we need is version string.
+	return espeakDLL.espeak_Info(None)
```
Since this function returns bytes, and the only place in which it is used is the message in the log stating which version of eSpeak is in use have you considered decoding this to avoid leading "b" being shown in the log viewer i.e.:
`return espeakDLL.espeak_Info(None).decode()`

josephsl commented:
Hi, should we also remove encoding statement at the top of the file? If yes, I’ll do that while addressing comments. As for encoding from bytes to str, I’ll do that once Grep tells me it is safe to do so. Thanks.

josephsl committed:
Synth drivers/eSpeak NG internal: address review comments.

Comment from Lukasz Golonka: remove file name from copyright header, use bytes.decode to transform eSpeak NG version string from bytes to Unicode.
